homeassistant/components/niko_home_control/light.py

```diff
@@ -1,53 +1,81 @@
 """Support for Niko Home Control."""
+from datetime import timedelta
 import logging
 
 import voluptuous as vol
 
+# Import the device class from the component that you want to support
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS, PLATFORM_SCHEMA, Light)
-from homeassistant.const import CONF_HOST
+from homeassistant.const import CONF_HOST, CONF_SCAN_INTERVAL
 from homeassistant.exceptions import PlatformNotReady
 import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
 
 _LOGGER = logging.getLogger(__name__)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+SCAN_INTERVAL = timedelta(seconds=30)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
 })
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
     """Set up the Niko Home Control light platform."""
     import nikohomecontrol
-
     host = config[CONF_HOST]
 
     try:
-        hub = nikohomecontrol.Hub({
+        nhc = nikohomecontrol.NikoHomeControl({
             'ip': host,
             'port': 8000,
-            'timeout': 20000,
-            'events': True
+            'timeout': 20000
         })
+        niko_data = NikoHomeControlData(hass, nhc)
+        await niko_data.async_update()
     except OSError as err:
         _LOGGER.error("Unable to access %s (%s)", host, err)
         raise PlatformNotReady
 
-    add_entities(
-        [NikoHomeControlLight(light, hub) for light in hub.list_actions()],
-        True)
+    async_add_entities([
+        NikoHomeControlLight(light, niko_data) for light in nhc.list_actions()
+        ], True)
 
 
 class NikoHomeControlLight(Light):
     """Representation of an Niko Light."""
 
-    def __init__(self, light, nhc):
+    def __init__(self, light, data):
         """Set up the Niko Home Control light platform."""
-        self._nhc = nhc
+        self._data = data
         self._light = light
+        self._unique_id = "light-{}".format(light.id)
         self._name = light.name
-        self._state = None
+        self._state = light.is_on
         self._brightness = None
+        _LOGGER.debug("Init new light: %s", light.name)
+
+    @property
+    def unique_id(self):
+        """Return unique ID for light."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
```

Comment:
`device_info` property is only read if the platform is loaded via config entry.

https://developers.home-assistant.io/docs/en/device_registry_index.html#defining-devices